### ISSUE TITLE
add "alwayscommit" option to annex add (and may be other) commands

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#127]
annex by default always commits right away, but that is unacceptable e.g. when I need to add many files one at a time (while creating a repository from the web).  That is why so far I was delaying commit in repos.py  with 
```python
self._default_annex_opts = ['-c', 'annex.alwayscommit=false']
```
so I guess we need an option like that for at least to `add` and `addurl*` ones so we could prepare fully for an atomic single commit ;-)

**Comment by bpoldrack** [comment]
Right - on my todo-list already.

**Comment by yarikoptic** [comment]
was addressed by #143
